castro/__init__.py

```diff
@@ -114,19 +129,32 @@ def encode(self):
             scale = "-qscale 0"
         else:
             scale = "-sameq"
+
+        newsize = ""
+        if downscale:
+            DOWNSCALE_WIDTH = 1024
+            w, h = self._get_size(self.filepath)
+            log.info("Performing downscaling, for %sx%s video", w, h)
+            if w and h and w > DOWNSCALE_WIDTH:
+                new_w = DOWNSCALE_WIDTH
+                new_h = int((new_w / float(2560)) * 1600)
+                log.info("Size after downscaling %sx%s", new_w, new_h)
+                newsize = "-s %sx%s " % (new_w, new_h)
+
         print "Running ffmpeg: encoding and creating keyframes"
-        cmd = "ffmpeg -y -i %s -g %s " + scale + " %s"
+        cmd = "ffmpeg -y -i %s -g %s " + scale + " %s%s"
         if self.h264:
             cmd = ("ffmpeg -y -i %s -vcodec libx264 -coder 0 -flags -loop"
                    " -cmp +chroma -partitions -parti8x8-parti4x4-partp8x8-partb8x8"
                    " -me_method dia -subq 0 -me_range 16 -g %s -keyint_min 25"
                    " -sc_threshold 0 -i_qfactor 0.71 -b_strategy 0 -qcomp 0.6"
                    " -qmin 10 -qmax 51 -qdiff 4 -bf 0 -refs 1 -directpred 1 -trellis 0"
                    " -flags2 -bpyramid-mixed_refs-wpred-dct8x8+fastpskip-mbtree -wpredp"
-                   " 0 -aq_mode 0 -crf 30 %s")
+                   " 0 -aq_mode 0 -crf 30 %s%s")
 
         os.system(cmd % (self.filepath,
                          self.framerate * self.seconds_per_keyframe,
+                         newsize,
```

Comment:
I'm not 100% sure what are allowed params here, will try resolution reduction first :)